wdl-lint/src/rules/blanks_between_elements.rs

```diff
@@ -0,0 +1,546 @@
+                } else {
+                    let first = decl
+                        .syntax()
+                        .prev_sibling()
+                        .is_some_and(|f| f.kind() == SyntaxKind::InputSectionNode);
+
+                    let prev = skip_preceding_comments(decl.syntax());
+
+                    if first {
+                        check_prior_spacing(&prev, state, true, false);
+                    }
+                }
+            }
+        }
+    }
+
+    fn bound_decl(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        decl: &wdl_ast::v1::BoundDecl,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        let actual_start = skip_preceding_comments(decl.syntax());
+
+        let prior = actual_start.prev_sibling_or_token();
+        if let Some(p) = prior {
+            if p.kind() == SyntaxKind::Whitespace {
+                let count = p.to_string().chars().filter(|c| *c == '\n').count();
+                // If we're in an `input` or `output`, we should have no blank lines, so only
+                // one `\n` is allowed.
+                if self.state == State::InputSection || self.state == State::OutputSection {
+                    if count > 1 {
+                        state.add(excess_blank_line(p.text_range().to_span()));
+                    }
+                } else {
+                    let first = decl
+                        .syntax()
+                        .prev_sibling()
+                        .is_some_and(|f| f.kind() == SyntaxKind::InputSectionNode);
+
+                    let prev = skip_preceding_comments(decl.syntax());
+
+                    if first {
+                        check_prior_spacing(&prev, state, true, false);
+                    }
+                }
+            }
+        }
+    }
+
+    fn conditional_statement(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        stmt: &wdl_ast::v1::ConditionalStatement,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        let first = stmt
+            .syntax()
+            .prev_sibling()
+            .is_some_and(|f| f.kind() == SyntaxKind::InputSectionNode);
+
+        let prev = skip_preceding_comments(stmt.syntax());
+
+        if first {
+            check_prior_spacing(&prev, state, true, false);
+        }
+    }
+}
+
+/// Check if the given syntax node is the first element in the block.
+fn is_first_element(syntax: &SyntaxNode) -> bool {
+    let mut prev = syntax.prev_sibling_or_token();
+    while let Some(ref cur) = prev {
+        match cur {
+            NodeOrToken::Token(t) => {
+                if t.kind() == SyntaxKind::OpenBrace {
+                    return true;
+                }
+            }
+            NodeOrToken::Node(_) => {
+                return false;
+            }
+        }
+        prev = cur.prev_sibling_or_token();
+    }
+    true
```

Comment:
I was kicking this around in my head, but I think I'd prefer to go with `unreachable!()` here. The method assumes that the element being examined is within a block. The only way a block is entered right now is with an `OpenBrace`. If that ever changes in the future, I would want it to be flagged rather than implicitly assumed that the element is first.

Reply:
It is used for `workflow` and `task`. So it doesn't assume it is within an element. Here `block` could be the document `root`.

Reply:
I suppose it is `unreachable!()` because there must be a `VersionStatement` to open the document. So even in those cases, it can't reach that value.